Failures in ParentChildClientYamlTestSuiteIT

There are failures in three tests in this test suite. Excerpts and repro lines for one of them are below, but all are available in the linked build scans.

**Build scan**:
https://gradle-enterprise.elastic.co/s/5flx6k6jaabhu
https://gradle-enterprise.elastic.co/s/2y4jh76yjt3ti

**Repro line**:
`./gradlew ':modules:parent-join:yamlRestTest' --tests "org.elasticsearch.join.ParentChildClientYamlTestSuiteIT.test {yaml=/50_order_by/order by sub agg containing join}" -Dtests.seed=5DFF4DD8A9ED1A44 -Dtests.security.manager=true -Dtests.locale=nl-NL -Dtests.timezone=Etc/GMT+8 -Druntime.java=8	
`

**Reproduces locally?**: Yes

**Applicable branches**: 7.11

**Failure history**: Appears to have started on 3/10/21 with the release of 7.11.2

**Failure excerpt**:

```
java.lang.AssertionError: Failure at [/50_order_by:68]: aggregations.name.buckets.0.key didn't match expected value: |  
-- | --
  | aggregations.name.buckets.0.key: expected String [pig] but was String [cow]
```

--- a/modules/parent-join/src/yamlRestTest/resources/rest-api-spec/test/50_order_by.yml
+++ b/modules/parent-join/src/yamlRestTest/resources/rest-api-spec/test/50_order_by.yml
@@ -1,8 +1,8 @@
 ---
 "order by sub agg containing join":
   - skip:
-      reason: "It was fixed it 7.12.0"
-      version: " - 7.11.99"
+      reason: "https://github.com/elastic/elasticsearch/issues/66876"
+      version: "7.11.1 - "
   - do:
       indices.create:
           index: test_1
@@ -72,8 +72,8 @@
 ---
 "order by sub agg containing join and nested":
   - skip:
-      reason: "It was fixed it 7.12.0"
-      version: " - 7.11.99"
+      reason: "https://github.com/elastic/elasticsearch/issues/66876"
+      version: "7.11.1 - "
   - do:
       indices.create:
           index: test_1
@@ -152,8 +152,8 @@
 ---
 "order by sub agg containing join and nested and filter":
   - skip:
-      reason: "It was fixed it 7.12.0"
-      version: " - 7.11.99"
+      reason: "https://github.com/elastic/elasticsearch/issues/66876"
+      version: "7.11.1 - "
   - do:
       indices.create:
           index: test_1

